### PR TITLE
Move S.IO.FileAccess to corelib.

### LINKED ISSUE #12169: Push System.IO.FileAccess enum into corelib
Required by UnmanagedMemoryStream required by ResourceManager.GetString

### PR DESCRIPTION
**Pull request by alexperovich**
This enum is used by UnmanagedMemoryStream which is required for ResourceManager.GetStream()

Fixes #12169

### CODE REVIEW

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj [hunk 1]
```diff
@@ -4,24 +4,23 @@
...
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Release|AnyCPU'" />
+  <ItemGroup>
+    <TargetingPackReference Include="mscorlib" />
+  </ItemGroup>
   <ItemGroup Condition="'$(TargetGroup)' != 'net46'">
-    <Compile Include="System\IO\FileAccess.cs" />
```
Comment (stephentoub) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: If nothing else still uses the file, it should be deleted, too.

Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: Deleted

**Comment by alexperovich** [comment]
@weshaggard

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj [hunk 2]
```diff
@@ -4,24 +4,23 @@
...
-    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'net46'">true</IsPartialFacadeAssembly>
+    <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
     <NuGetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.3</NuGetTargetMoniker>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Release|AnyCPU'" />
+  <ItemGroup>
+    <TargetingPackReference Include="mscorlib" />
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: This change is going to clash with https://github.com/dotnet/corefx/pull/12187, where @jkotas is moving everything to build against System.Private.CoreLib.

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj [hunk 3]
```diff
@@ -4,24 +4,23 @@
   <PropertyGroup>
     <ProjectGuid>{6C05678E-394C-4CFF-B453-A18E28C8F2C3}</ProjectGuid>
     <AssemblyName>System.IO.FileSystem.Primitives</AssemblyName>
-    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'net46'">true</IsPartialFacadeAssembly>
+    <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
     <NuGetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.3</NuGetTargetMoniker>
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: You need to bump the assembly version and the target framework. for this library. See my docs which are still in PR https://github.com/dotnet/corefx/pull/12036, and please add feedback to that PR if you have further questions.

Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: I have bumped the versions

**Comment by alexperovich** [comment]
@weshaggard PTAL

File: src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds [hunk 4]
```diff
@@ -4,6 +4,10 @@
   <ItemGroup>
     <Project Include="System.IO.FileSystem.Primitives.Tests.csproj"/>
     <Project Include="System.IO.FileSystem.Primitives.Tests.csproj">
+      <TargetGroup>netstandard1.7</TargetGroup>
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: @joperezr this will conflict with your PR.

File: src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds [hunk 5]
```diff
@@ -4,7 +4,11 @@
   <ItemGroup>
     <Project Include="System.IO.FileSystem.Primitives.Tests.csproj"/>
     <Project Include="System.IO.FileSystem.Primitives.Tests.csproj">
-      <TestTFMs>netcore50;net46</TestTFMs>
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: why did you remove the netcore50 test configuration? I would expect we should still make sure we test that configuration.

File: src/System.IO.FileSystem.Primitives/dir.props [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.0.2.0</AssemblyVersion>
+    <AssemblyVersion>4.0.3.0</AssemblyVersion>
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/dir.props]: We generally bump minor versions but this is just moving one type down. @ericstj what is your thoughts on version bumping in this case?

File: src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds (hunk 5 above)
Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: I'll put it back.

**Comment by alexperovich** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Release Build and Test please (Build Timeout)

File: src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds [hunk 7]
```diff
@@ -4,7 +4,11 @@
...
+      <TargetGroup>netstandard1.7</TargetGroup>
+      <TestTFMs>netcoreapp1.1</TestTFMs>
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: Looks like @joperezr PR got in before yours. Can you please adjust this .builds file and csproj such that the netstandard1.7 is the default configuration. See https://github.com/dotnet/corefx/pull/12241 as an example

Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: Sure

File: src/System.IO.FileSystem.Primitives/tests/project.json [hunk 8]
```diff
@@ -13,14 +13,12 @@
     "Microsoft.DotNet.BuildTools.TestSuite": "1.0.0-prerelease-00807-03"
   },
   "frameworks": {
-    "netstandard1.3": {}
+    "netstandard1.3": {},
+    "netstandard1.7": {}
   },
   "supports": {
-    "coreFx.Test.netcore50": {},
```
Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/tests/project.json]: I put it back

**Comment by alexperovich** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test please (disk IO issue)

**Comment by alexperovich** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test please (disk IO issue)

File: src/System.IO.FileSystem.Primitives/dir.props (hunk 6 above)
Comment (ericstj) [inline src/System.IO.FileSystem.Primitives/dir.props]: If all that was changed was impl then we don't need to change version at all (I typically do that at the start of a release).

File: src/System.IO.FileSystem.Primitives/src/project.json [hunk 9]
```diff
@@ -1,12 +1,14 @@
...
+      }
+    },
     "netstandard1.3": {
       "dependencies": {
-        "System.Runtime": "4.0.20"
-      },
-      "imports": [
-        "dotnet5.4"
-      ]
+        "Microsoft.TargetingPack.Private.CoreCLR": "1.2.0-beta-24603-02"
```
Comment (joperezr) [inline src/System.IO.FileSystem.Primitives/src/project.json]: >"Microsoft.TargetingPack.Private.CoreCLR": "1.2.0-beta-24603-02" [](start = 8, length = 64)

Why did we make this change to bind to coreclr instead of going through System.Runtime like we did before?

File: src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds (hunk 5 above)
Comment (ericstj) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: Moreover, why'd you change this at all?

Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: I didn't catch this when I changed this back. Will fix.

File: src/System.IO.FileSystem.Primitives/tests/project.json [hunk 10]
```diff
@@ -18,9 +18,7 @@
   "supports": {
     "coreFx.Test.netcore50": {},
     "coreFx.Test.netcoreapp1.0": {},
-    "coreFx.Test.net46": {},
-    "coreFx.Test.net461": {},
-    "coreFx.Test.net462": {},
+    "coreFx.Test.netcoreapp1.1": {},
```
Comment (ericstj) [inline src/System.IO.FileSystem.Primitives/tests/project.json]: Why was this required?

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj [hunk 11]
```diff
@@ -4,24 +4,31 @@
...
+    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' != 'netcore50aot'">true</IsPartialFacadeAssembly>
     <NuGetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.3</NuGetTargetMoniker>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'net46_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Release|AnyCPU'" />
+  <ItemGroup Condition="'$(TargetGroup)' == ''">
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: nit: can you please include with the other reference below.

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.builds [hunk 12]
```diff
@@ -6,6 +6,9 @@
     <Project Include="System.IO.FileSystem.Primitives.csproj">
       <TargetGroup>net46</TargetGroup>
     </Project>
+    <Project Include="System.IO.FileSystem.Primitives.csproj">
+      <TargetGroup>netcore50aot</TargetGroup>
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.builds]: I think this needs to be netcore50

File: src/System.IO.FileSystem.Primitives/tests/project.json (hunk 10 above)
Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/tests/project.json]: This is required because netcoreapp1.1 is the default TestTFM now and the project file doesn't change that.

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.builds (hunk 12 above)
Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.builds]: Changed

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj [hunk 13]
```diff
@@ -4,24 +4,31 @@
   <PropertyGroup>
     <ProjectGuid>{6C05678E-394C-4CFF-B453-A18E28C8F2C3}</ProjectGuid>
     <AssemblyName>System.IO.FileSystem.Primitives</AssemblyName>
-    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'net46'">true</IsPartialFacadeAssembly>
+    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' != 'netcore50aot'">true</IsPartialFacadeAssembly>
```
Comment (jkotas) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: I do not think we should be bothering with keeping the netcore50 build live. Delete it instead - it is what we are doing everywhere else.

File: src/System.IO.FileSystem.Primitives/src/project.json (hunk 9 above)
Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/src/project.json]: It does not.

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj (hunk 13 above)
Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: We cannot remove this netcore50 build. This package was portable and now is not portable to .net native.

**Comment by alexperovich** [comment]
@dotnet-bot test this please

File: src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj (hunk 13 above)
Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: There isn't a netcore50 binary in the previous stable package

Comment (jkotas) [inline src/System.IO.FileSystem.Primitives/src/System.IO.FileSystem.Primitives.csproj]: Then just bump the default build to netstandard1.7 like I am doing it here: https://github.com/dotnet/corefx/pull/12187.

File: src/System.IO.FileSystem.Primitives/tests/project.json (hunk 10 above)
Comment (alexperovich) [inline src/System.IO.FileSystem.Primitives/tests/project.json]: This file is unchanged now

**Comment by alexperovich** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Debug Build and Test please (Build Timeout)

File: src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds (hunk 5 above)
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: We should still keep netcore50 as a TestTFM.

Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.builds]: Which would mean you still need a netstandard1.3 build configuration for this project.

File: src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.csproj [hunk 14]
```diff
@@ -7,7 +7,7 @@
     <ProjectGuid>{2EF7EFA5-F171-4CAB-8A29-32833949FD87}</ProjectGuid>
     <OutputType>Library</OutputType>
     <AssemblyName>System.IO.FileSystem.Primitives.Tests</AssemblyName>
-    <NugetTargetMoniker>.NETStandard,Version=v1.3</NugetTargetMoniker>
+    <NugetTargetMoniker>.NETStandard,Version=v1.7</NugetTargetMoniker>
```
Comment (weshaggard) [inline src/System.IO.FileSystem.Primitives/tests/System.IO.FileSystem.Primitives.Tests.csproj]: Once you add the netstandard1.3 build configuration you will need to conditions this on an empty targetgroup.
